browser/brave_content_browser_client.cc

```diff
@@ -421,6 +421,16 @@ void BraveContentBrowserClient::
       RegisterAssociatedInterfaceBindersForRenderFrameHost(render_frame_host,
                                                            associated_registry);
 }
+void BraveContentBrowserClient::RegisterWebUIInterfaceBrokers(
+    content::WebUIBrowserInterfaceBrokerRegistry& registry) {
+  ChromeContentBrowserClient::RegisterWebUIInterfaceBrokers(registry);
+#if BUILDFLAG(ENABLE_BRAVE_VPN) && !BUILDFLAG(IS_ANDROID)
+  if (brave_vpn::IsBraveVPNEnabled()) {
```

Comment:
it definitely makes sense to me that this should go here, but shouldn't wallet and anything else that is a webui popup go here?

Reply:
You know what, I'm not sure. In chromium [they still have both ways](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.h;l=34-45;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1) of registering the mojom interface for a WebUI class.

```
// [PopulateChromeWebUIFrameBinders](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=37?q=RegisterWebUIControllerInterfaceBinder&gsn=PopulateChromeWebUIFrameBinders&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dsrc%2Fchrome%2Fbrowser%2Fchrome_browser_interface_binders.h%238po985FhCkc7I0ellqaH_m_ndXiSx6g1VpcRGc8GHTw)() registers BrowserInterfaceBroker's
// GetInterface() handler callbacks for chrome-specific document-scoped
// interfaces used from WebUI pages (e.g. chrome://bluetooth-internals).
void PopulateChromeWebUIFrameBinders(
    mojo::[BinderMapWithContext](https://source.chromium.org/chromium/chromium/src/+/main:mojo/public/cpp/bindings/binder_map.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=37)<content::[RenderFrameHost](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/render_frame_host.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=123)*>* [map](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=38?q=RegisterWebUIControllerInterfaceBinder&gsn=map&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dsrc%2Fchrome%2Fbrowser%2Fchrome_browser_interface_binders.h%23NBz6KkqMO8Gjx0mKBKKg1M33v4h_CIS4zTwV-ecKVGE),
    content::[RenderFrameHost](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/render_frame_host.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=123)* [render_frame_host](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=39?q=RegisterWebUIControllerInterfaceBinder&gsn=render_frame_host&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dsrc%2Fchrome%2Fbrowser%2Fchrome_browser_interface_binders.h%23_WvlRMimRJSjU-gkosA2HM7iqyXTMCgrfgVT69Tzr34));

// [PopulateChromeWebUIFrameInterfaceBrokers](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=44?q=RegisterWebUIControllerInterfaceBinder&gsn=PopulateChromeWebUIFrameInterfaceBrokers&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dsrc%2Fchrome%2Fbrowser%2Fchrome_browser_interface_binders.h%23FV-95cosGIejgJUGFOLU4K1tj_gt_nBMmLtXt58843Y) registers BrowserInterfaceBrokers
// for each WebUI, these brokers are used to handle that WebUI's JavaScript
// Mojo.bindInterface calls.
void PopulateChromeWebUIFrameInterfaceBrokers(
    content::[WebUIBrowserInterfaceBrokerRegistry](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_ui_browser_interface_broker_registry.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=60)& [registry](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.h;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1;l=45?q=RegisterWebUIControllerInterfaceBinder&gsn=registry&gs=kythe%3A%2F%2Fchromium.googlesource.com%2Fchromium%2Fsrc%3Flang%3Dc%252B%252B%3Fpath%3Dsrc%2Fchrome%2Fbrowser%2Fchrome_browser_interface_binders.h%23K5THRdfgzyvnf9RBLNlO_z5kEw8f6DvozStwWnyHPnA));
 ```

I'm not entirely sure what the implications are for the 2 different ways of doing that. I just saw that they had used the "registry" way for a couple [recent Untrusted WebUI](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/chrome_browser_interface_binders.cc;l=1187;drc=0dc049afd381cb2d2d209f4b6467f6dce677a870;bpv=1;bpt=1)s, and the API seemed nicer, so I used it for this (also to make the first use of the relevant override).

Reply:
I'm not entirely sure either, I was hoping you were :) Sounds like it needs some investigation